include/matx/core/tensor_impl.h

```diff
@@ -1080,9 +1106,54 @@ IGNORE_WARNING_POP_GCC
       ldata_ = data;
     }
 
+    template <typename Task>
+    __MATX_INLINE__ void apply_dep_to_task(Task &&task, int perm) const noexcept
+    {
+      auto &ld = stf_ldata_->value();
+
+      using namespace cuda::experimental::stf;
+#if 0
+      data_place place = getDataPlace(Data());
+#endif
+
+      if (perm == 0) {
+          task.add_deps(ld.write());
```

Comment:
We could directly build a task_dep in CUDASTF, matching perm value with the type ... But it seems there is no such thing as a clean way to do this !